analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -2271,7 +2271,7 @@ public void FP_Increment(List<int> list)
             var steps = 0;
             while (list.Any())
             {
-                if (steps++ > MaxStepCount) // Noncompliant FP
+                if (steps++ > MaxStepCount) // Compliant
```

Comment:
Let's add test cases on:
- boundary conditions
- nested loops

They will prove useful when the loop handling is further improved in the future.
e.g.

```cs
void ForLoop_Regular()
{
    for (int i = 0; i < 10; i++)
    {
        _ = i > -1 ? 0 : 42;        // Noncompliant
        _ = i >= -1 ? 0 : 42;       // Noncompliant
        _ = i > 0 ? 0 : 42;         // Compliant
        _ = i >= 0 ? 0 : 42;        // Noncompliant
        _ = i > 9 ? 0 : 42;         // Noncompliant
        _ = i >= 9 ? 0 : 42;        // Compliant
        _ = i > 10 ? 0 : 42;        // Noncompliant
        _ = i >= 10 ? 0 : 42;       // Noncompliant
        _ = i >= 10 ? 0 : 42;       // Noncompliant
        _ = i >= i + 1 ? 0 : 42;    // FN
    }
}

void ForLoop_IncreaseInside()
{
    for (int i = 0; i < 10;)
    {
        _ = i > -1 ? 0 : 42;        // Noncompliant
        _ = i >= -1 ? 0 : 42;       // Noncompliant
        _ = i > 0 ? 0 : 42;         // Compliant
        _ = i >= 0 ? 0 : 42;        // Noncompliant
        _ = i > 9 ? 0 : 42;         // Noncompliant
        _ = i >= 9 ? 0 : 42;        // Compliant
        _ = i > 10 ? 0 : 42;        // Noncompliant
        _ = i >= 10 ? 0 : 42;       // Noncompliant
        _ = i >= 10 ? 0 : 42;       // Noncompliant
        _ = i >= i + 1 ? 0 : 42;    // FN
        i++;
    }
}

void ForLoop_Nested()
{
    for (int i = 0; i < 10; i++)
        for (int j = i + 1; j < i + 10; j++)
        {
            _ = j >= 0 ? 0 : 42;        // Noncompliant
            _ = j > 0 ? 0 : 42;         // Noncompliant
            _ = j < 18 ? 0 : 42;        // Compliant
            _ = j < 19 ? 0 : 42;        // Noncompliant
            _ = i + j < 17 ? 0 : 42;    // Compliant
            _ = i + j < 28 ? 0 : 42;    // Noncompliant
            _ = i != j ? 0 : 42;        // FN
            _ = i > j ? 0 : 42;         // FN
        }
}

void ForLoop_Nested_IncrementInside()
{
    for (int i = 0; i < 10;)
    {
        for (int j = i + 1; j < 10;)
        {
            _ = j >= 0 ? 0 : 42;        // Noncompliant
            _ = j > 0 ? 0 : 42;         // Noncompliant
            _ = j < 18 ? 0 : 42;        // Compliant
            _ = j < 19 ? 0 : 42;        // Noncompliant
            _ = i + j < 17 ? 0 : 42;    // Compliant
            _ = i + j < 28 ? 0 : 42;    // Noncompliant
            _ = i != j ? 0 : 42;        // FN
            _ = i > j ? 0 : 42;         // FN
            j++;
        }
        i++;
    }
}

void ForLoop_MultipleLoopVariables()
{
    for (int i = 0, j = 1; i < 10; i++, j++)
    {
        _ = i >= 0 ? 0 : 42;            // Noncompliant
        _ = j > 0 ? 0 : 42;             // Noncompliant
        _ = j < 3 ? 0 : 42;             // Noncompliant - FP
        _ = i + j > 0 ? 0 : 42;         // Noncompliant
        _ = i + j > 1 ? 0 : 42;         // Compliant
    }
}

void ForLoop_ZeroOrOneExecutions()
{
    for (int i = 0; i < 0; i++)
    {
        _ = i > 0 ? 0 : 42;             // Compliant
    }

    for (int i = 0; i < 1; i++)
    {
        _ = i > 0 ? 0 : 42;             // Noncompliant
    }

    for (int i = 0; i < 1; ++i)
    {
        _ = i > 0 ? 0 : 42;             // Noncompliant
    }
}
```

Reply:
Very nice test cases, thank you!